libtiny_client/src/state.rs

```diff
@@ -232,6 +322,73 @@ impl StateInner {
+                        .unwrap();
+                    // Get channel name from params
+                    if self.nickserv_ident.is_some() {
+                        // Helper for creating an event
+                        let create_message = |msg: String| Event::ChannelJoinError {
+                            chan: channel.clone(),
+                            msg,
+                        };
+                        // Find channel in self.chans
+                        if let Some(idx) = utils::find_idx(&self.chans, |c| c.name == *channel) {
+                            let chan = &mut self.chans[idx];
+                            // Retry joining channel if retries are available
+                            if let Some(retries) = chan.retry_join() {
+                                let retry_msg = format!(
+                                    "Attempting to rejoin {} in 10 seconds... ({}/{})",
+                                    channel,
+                                    MAX_JOIN_RETRIES - retries,
+                                    MAX_JOIN_RETRIES
+                                );
+                                snd_ev.try_send(create_message(retry_msg)).unwrap();
+                                let snd_irc_msg = snd_irc_msg.clone();
+                                // Spawn task and delay rejoin to give NickServ time to identify nick
+                                let (snd_abort, rcv_abort) = tokio::sync::mpsc::channel::<()>(1);
+                                match &mut chan.join_state {
+                                    JoinState::NotJoined => chan.set_joining(snd_abort),
+                                    JoinState::Joining { stop_task, .. } => *stop_task = snd_abort,
+                                    JoinState::Joined => {
+                                        error!("Unexpected JoinState for channel.");
+                                        return;
+                                    }
+                                }
+                                tokio::task::spawn_local(retry_channel_join(
+                                    channel.clone(),
+                                    snd_irc_msg,
+                                    rcv_abort,
+                                ));
+                            } else {
+                                // No more retries
+                                let no_retries_msg = format!("Unable to join {}.", channel);
+                                snd_ev.try_send(create_message(no_retries_msg)).unwrap();
+                            }
+                        } else {
+                            warn!("Could not find channel in server state channel list.");
+                        }
+                    } else {
+                        warn!("Received 477 reply but nickserv_ident is not configured.");
+                    }
+                } else {
+                    warn!("Received 477 reply but could not parse parameters");
```

Comment:
Let's show the message format here:

```suggestion
                    warn!("Could not parse 477 reply: {:?}", cmd);
```